apps/taxes/src/modules/avatax/tax-code/avatax-tax-codes.router.ts

```diff
@@ -0,0 +1,30 @@
+import { router } from "../../trpc/trpc-server";
+
+import { z } from "zod";
+import { createLogger } from "../../../lib/logger";
+import { protectedClientProcedure } from "../../trpc/protected-client-procedure";
+import { AvataxConnectionService } from "../configuration/avatax-connection.service";
+import { AvataxTaxCodesService } from "./avatax-tax-codes.service";
+
+const getAllForIdSchema = z.object({ connectionId: z.string() });
+
+export const avataxTaxCodesRouter = router({
+  getAllForId: protectedClientProcedure.input(getAllForIdSchema).query(async ({ ctx, input }) => {
+    const logger = createLogger({
+      location: "avataxTaxCodesRouter.getAllForId",
+    });
+
+    const connectionService = new AvataxConnectionService(
+      ctx.apiClient,
+      ctx.appId!,
+      ctx.saleorApiUrl
```

Comment:
if client is passed, why you need also app ID and API URL?

Reply:
`client` and `appId` are arguments of `createSettingsManager` and `saleorApiUrl` is an argument of methods of the metadata manager, e.g. `.get(..., saleorApiUrl)`.